tests/ctst/common/common.ts

```diff
@@ -15,3 +17,31 @@ Given('a {string} bucket', async function (versioning) {
         await S3.putBucketVersioning(this.getCommandParameters());
     }
 });
+
+Given('an existing bucket {string} {string} versioning', async function (bucketName, hasVersioning) {
+    this.resetCommand();
+    const preName = this.parameters.AccountName || Constants.ACCOUNT_NAME;
+    bucketName = bucketName || `${preName}${Constants.BUCKET_NAME_TEST}${Utils.randomString()}`.toLocaleLowerCase();
+    this.saved.bucketName = bucketName;
+    this.addCommandParameter({ bucket: bucketName });
+    await S3.createBucket(this.getCommandParameters());
+    if (hasVersioning === 'with') {
+        this.addCommandParameter({ versioningConfiguration: `Status=Enabled`});
+        await S3.putBucketVersioning(this.getCommandParameters());
+    }
+});
+
+Given('an object that {string} {string} version-Id',
+    async function (objectExists, withVersioning) {
+    this.resetCommand();
+    if (objectExists === 'exists') {
+        this.saved.objectName = 'x'.repeat(10);
+        this.addCommandParameter({key: this.saved.objectName});
+        this.addCommandParameter({bucket: this.saved.bucketName});
+        if (withVersioning === 'with') {
+            this.saved.versionId = '3sL4kqtJlcpXroDTDmJ+rmSpXd3dIbrHY+MTRCxf3vjVBH40Nr8X8gdRQBpUMLUo';
+            this.addCommandParameter({versionId: this.saved.versionId});
+        }
```

Comment:
Should be removed, we cannot specify a version ID when putting an object, this is done by the backend. If the bucket is version-enabled, it will be returned by the first command.

Reply:
Closing as resoved in latest commit.